pusher-platform-core/src/main/kotlin/com/pusher/platform/ErrorResolver.kt

```diff
@@ -25,37 +28,48 @@ fun ErrorResponse.isRetryable(): Boolean =
 
 
 class ErrorResolver(
-    private val connectivityHelper: ConnectivityHelper,
     private val retryOptions: RetryStrategyOptions,
-    private val scheduler: Scheduler,
     private val retryUnsafeRequests: Boolean = false
 ) {
 
     private var currentRetryCount = 0
     private var currentBackoffMillis = 0L
 
-    private val runningJobs = LinkedList<ScheduledJob>()
+    private val runningJobs = LinkedList<Future<*>>()
 
     fun resolveError(error: Error, callback: RetryStrategyResultCallback){
         when(error){
             is NetworkError -> {
-                connectivityHelper.onConnected { callback(Retry())}
+                if(retryOptions.limit < 0 || currentRetryCount <= retryOptions.limit){
+                    currentBackoffMillis = increaseCurrentBackoff()
+                    currentRetryCount += 1
```

Comment:
Can current backoff be derived directly from this?

Reply:
Thanks for mentioning this. I had a plan for this class but with the failing tests, I forgot about it. The new implementation uses that and is simplified drastically.